### PR TITLE
#118275 storing technical logs

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: logging/src/main/kotlin/com/ritense/logging/domain/LogginEvent.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import jakarta.persistence.Table
+import org.springframework.data.annotation.Immutable
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+@Immutable
+@Entity
+@Table(name = "logging_event")
+class LoggingEvent(
```
Comment (marijnritense) [inline logging/src/main/kotlin/com/ritense/logging/domain/LogginEvent.kt]: TItle of the class doesn't seem to match the filename. Could you change the filename? Same goes for the other domain classes.

File: logging/src/main/resources/config/logging/logback/valtimo-database-appender.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            </dataSource>
+        </connectionSource>
+        <insertHeaders>true</insertHeaders>
+    </appender>
+
+    <appender name="VALTIMODBASYNC" class="ch.qos.logback.classic.AsyncAppender">
+        <appender-ref ref="VALTIMODB" />
+        <includeCallerData>false</includeCallerData>
+        <maxFlushTime>5000</maxFlushTime>
+        <neverBlock>true</neverBlock>
```
Comment (marijnritense) [inline logging/src/main/resources/config/logging/logback/valtimo-database-appender.xml:48]: Leaving this as a note, no need to do anything with it right now.

We should probably discuss if async is the way to go. Performance wise this is probably a great idea, but that means if a log cant be stored, then we don't have a papertrail. Bit of a weird situation. but I think sticking with async is the right call for now.

File: logging/src/main/resources/config/liquibase/changelog/logback/mysql.sql [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Logback: the reliable, generic, fast and flexible logging framework.
```
Comment (ivo-ritense) [inline logging/src/main/resources/config/liquibase/changelog/logback/mysql.sql:1]: How does the license work in this case? Can we just include licensed code in our repo? Alternatively, is this included in the logback artifact and can we reference that?

Comment (Klaas-Ritense) [reply logging/src/main/resources/config/liquibase/changelog/logback/mysql.sql:1]: I've checked it and its allowed. 

It's not possible to reference the existing files in their library, due to small but necessary changed made to these files.

File: logging/src/main/kotlin/com/ritense/logging/domain/LogginEvent.kt (hunk 1 above)
Comment (Klaas-Ritense) [reply logging/src/main/kotlin/com/ritense/logging/domain/LogginEvent.kt]: Changed

File: logging/src/main/resources/config/logging/logback/valtimo-database-appender.xml (hunk 2 above)
Comment (Klaas-Ritense) [reply logging/src/main/resources/config/logging/logback/valtimo-database-appender.xml:48]: Luckily, this logback file creates 2 appenders: `VALTIMODB` and `VALTIMODBASYNC`. 

But yes, the `VALTIMODBASYNC` performance is much better

**Review by ivo-ritense — APPROVED** [review]

**Pull request merged**